contactaccessrights/CRM/Contactaccessrights/DAO/Rights.php

```diff
@@ -101,107 +71,112 @@ class CRM_Contactaccessrights_DAO_Rights extends CRM_Core_DAO
   /**
    * ID of the entity the permission grants access to (basically, option value ID).
    *
-   * @var string
+   * @var int
    */
   public $entity_id;
   /**
-   * class constructor
-   *
-   * @return civicrm_contactaccessrights_rights
+   * Class constructor.
    */
-  function __construct()
-  {
+  function __construct() {
     $this->__table = 'civicrm_contactaccessrights_rights';
     parent::__construct();
   }
   /**
-   * Returns foreign keys and entity references
+   * Returns foreign keys and entity references.
    *
    * @return array
    *   [CRM_Core_Reference_Interface]
    */
-  static function getReferenceColumns()
-  {
-    if (!self::$_links) {
-      self::$_links = static ::createReferenceColumns(__CLASS__);
-      self::$_links[] = new CRM_Core_Reference_Basic(self::getTableName() , 'contact_id', 'civicrm_contact', 'id');
+  static function getReferenceColumns() {
+    if (!isset(Civi::$statics[__CLASS__]['links'])) {
```

Comment:
This method (and a few others) seems to have been updated in the autogenerated DAOs, should it also be updated in the manually updated ones?

Reply:
Ideally they should, but honestly, I don't feel very comfortable applying these changes manually. The DAOs from Job Roles and Contracts have lots of customizations and some of them are not upgrade-safe (for example, the contract's DAOs all extend a Base custom DAO instead of CRM_Core_DAO). It's hard to be sure that nothing will break if we do this. So, since everything is working fine, for now, I prefer to not do this for now. Also, there will be some work to improve things in Job Contracts in the future (hopefully soon!) and that will be a better opportunity to fix this